modules/data.remote/inst/gee2pecan_l8.py

```diff
@@ -0,0 +1,191 @@
+"""
+Extracts Landsat 8 surface reflactance band data from Google Earth Engine and saves it in a netCDF file
+
+Requires Python3
+
+Bands retrieved: B1, B2, B3, B4, B5, B6, B7, B10, B11 along with computed NDVI
+
+If ROI is a Point, this function can be used for getting SR data from Landsat 7, 5 and 4 as well.
+
+Author: Ayush Prasad
+"""
+from gee_utils import create_geo, get_siteaoi, get_sitename
+import ee
+import pandas as pd
+import geopandas as gpd
+import datetime
+import os
+import xarray as xr
+import numpy as np
+import re
+
+ee.Initialize()
+
+
+def gee2pecan_l8(geofile, outdir, start, end, qc=1):
+    """
+    Extracts Landsat 8 SR band data from GEE
+
+    Parameters
+    ----------
+    geofile (str) -- path to the file containing the name and coordinates of ROI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date areaof the data request in the form YYYY-MM-DD
+
+    ic (str) -- image collection id of the Landsat SR product from Google Earth Engine
+
+    vi (bool) -- set to True if NDVI needs to be calculated
+
+    qc (bool) -- uses the cloud masking function if set to True
+
+    bands (list of str) -- bands to be retrieved. Default: B5, B4
+
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory. 
+ 
+    """
+
+    # scale (int) Default: 30
+    scale = 30
+    # bands retrieved
+    bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B10", "B11"]
```

Comment:
I'm surprised that all bands need to be listed out, just curious, isn't there some `ee.select()` function feature that gets all bands if nothing is specified explicitly? or of they have some function arg like `all=TRUE` that returns everything or something like that?  there isn't much you can do otherwise of course

Reply:
There isn't any function which does that. `.select()` exists but it's used for selecting the bands from the collection.